lightning/src/ln/offers_tests.rs

```diff
@@ -1070,6 +1070,266 @@ fn creates_refund_with_blinded_path_using_unannounced_introduction_node() {
 	}
 }
 
+/// Check that authentication fails when an invoice request is handled using the wrong context
+/// (i.e., was sent directly or over an unexpected blinded path).
```

Comment:
We are not testing the second point here, "or over an unexpected blinded path"

I think adding a test for it too will be a great idea!

Reply:
Updated to cover that case here and below.